ibis/backends/base/__init__.py

```diff
@@ -206,7 +207,65 @@ def _ipython_key_completions_(self) -> list[str]:
         return self._backend.list_tables()
 
 
-class BaseBackend(abc.ABC):
+# should have a better name
+class ResultHandler:
+    @staticmethod
+    def _import_pyarrow():
+        try:
+            import pyarrow
+        except ImportError:
+            raise ModuleNotFoundError(
+                "Exporting to arrow formats requires `pyarrow` but it is not installed"  # noqa: ignore
+            )
+        else:
+            return pyarrow
+
+    def to_pyarrow(
```

Comment:
I wonder, could we move this to `ibis.backends.pyarrow` somehow (fine in a follow-up too)?

Also, how could another backend, say polars register its own result types?

Reply:
We could move it, but I was planning (in a follow-up) to add a `to_pandas` method to the mixin.  However, it might be useful (to your point about e.g. polars registering a result type) to have a means for backends to register their own `to_*` methods.  Then the `to_pyarrow` would be defined in the `pyarrow` backend and `to_pandas` in the `pandas` backend.

Seems doable for the methods called via `backend.to_*` -- I'm not sure yet how we might also then add the relevant methods to expression objects

Reply:
Doing it in a follow up sounds good to me.